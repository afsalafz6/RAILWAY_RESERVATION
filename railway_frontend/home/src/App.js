import "./App.css";
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Home from "./Components/Home/Home";
import Register from "./Components/Register/Register";
import Login from "./Components/Login/Login";
import Traindetails from "./Components/Admin/TrainDetails/Traindetails";
import Dashboard from "./Components/Admin/Dashboard/Dashboard";
import Userdetails from "./Components/Admin/UserDetails/Userdetails";
import CreateTrain from "./Components/Admin/TrainDetails/CreateTrain";
import UserDashboard from "./Components/User/UserDashboard/UserDashboard";
import AllTrainList from "./Components/User/AllTrainList/AllTrainList";
import UserProfile from "./Components/User/UserProfile/UserProfile";
import BookTrain from "./Components/User/BookTrain/BookTrain";
import Payment from "./Components/User/Payment/Payment";
import BookingDetails from "./Components/Admin/BookingDetails/BookingDetails";
// import ETicket from "./Components/User/ETicket/ETicket";
import bookings from "./Components/User/Bookings/bookings";


function App() {
  return (
    <div>
      <Router>
        <Switch>
          <Route path='/' exact component={Home} />
          <Route path='/login' component={Login} />
          <Route path='/register' component={Register} />
          <Route path='/dashboard' exact component={Dashboard} />
          <Route path='/traindetails' exact component={Traindetails} />
          <Route path='/userdetails' exact component={Userdetails} />
          <Route path='/createtrain' exact component={CreateTrain} />
          <Route path='/createtrain/:trainid' exact component={CreateTrain} />
          <Route path='/usrdash' exact component={UserDashboard} />
          <Route path='/alltrain' exact component={AllTrainList} />
          <Route path='/profile' exact component={UserProfile} />
          <Route path='/booktrain/:trainid' exact component={BookTrain} />
          <Route path='/pay/:bookingid' exact component={Payment} />
          <Route path='/allbooking' exact component={BookingDetails} />
          {/* <Route path='/eticket' exact component={ETicket} /> */}
          <Route path='/bookings' exact component={bookings} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
