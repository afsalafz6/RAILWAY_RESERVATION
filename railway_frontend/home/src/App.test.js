// import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';

// test('renders App react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/Railway Reservation System/i);
//   expect(linkElement).toBeInTheDocument();
// });

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
 });