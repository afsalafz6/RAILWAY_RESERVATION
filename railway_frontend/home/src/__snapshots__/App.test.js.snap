// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div>
      <nav>
        <div
          className="nav-wrapper"
          style={
            Object {
              "backgroundColor": "#26a69a",
            }
          }
        >
          <a
            className="brand-logo"
            href="/"
          >
            <i
              className="material-icons circle white small"
            >
              train
            </i>
          </a>
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <a
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
            <li>
              <a
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div
      className="parallax-container"
      id="index-banner"
    >
      <div
        className="section no-pad-bot"
      >
        <div
          className="container"
        >
          <br />
          <br />
          <h1
            className="header center teal-text text-lighten-2"
          >
            Railway Reservation System
          </h1>
          <div
            className="row center"
          >
            <h5
              className="header col s12 light"
            >
              Public Transportation System Governed by Indian Government
            </h5>
          </div>
          <div
            className="row center"
          >
            <a
              className="btn-large waves-effect waves-light teal lighten-1"
              href="/register"
              id="download-button"
            >
              Register
            </a>
          </div>
          <br />
          <br />
        </div>
      </div>
      <div
        className="parallax"
      >
        <img
          alt="Railway img 1"
          src="images/background1.jpg"
        />
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="section"
      >
        <div
          className="row"
        >
          <div
            className="col s12 m4"
          >
            <div
              className="icon-block"
            >
              <h2
                className="center brown-text"
              >
                <i
                  className="material-icons"
                >
                  flash_on
                </i>
              </h2>
              <h5
                className="center"
              >
                Speeds up Reservation
              </h5>
              <p
                className="light"
              >
                We did most of the functionality provide by Indian Railway in the way of Ticket Reservation.
              </p>
            </div>
          </div>
          <div
            className="col s12 m4"
          >
            <div
              className="icon-block"
            >
              <h2
                className="center brown-text"
              >
                <i
                  className="material-icons"
                >
                  group
                </i>
              </h2>
              <h5
                className="center"
              >
                Available 24/7
              </h5>
              <p
                className="light"
              >
                you areable to book a Reservation across all over India at any time.
              </p>
            </div>
          </div>
          <div
            className="col s12 m4"
          >
            <div
              className="icon-block"
            >
              <h2
                className="center brown-text"
              >
                <i
                  className="material-icons"
                >
                  settings
                </i>
              </h2>
              <h5
                className="center"
              >
                Easy to work with
              </h5>
              <p
                className="light"
              >
                We have provided all require functionality with great user experience. all kind of user can register through this platform easily.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="parallax-container valign-wrapper"
    >
      <div
        className="section no-pad-bot"
      >
        <div
          className="container"
        >
          <div
            className="row center"
          >
            <h5
              className="header col s12 light"
            >
              Railway Booking make Easy
            </h5>
          </div>
        </div>
      </div>
      <div
        className="parallax"
      >
        <img
          alt="Unsplashed background img 2"
          src="images/background2.jpg"
        />
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="section"
      >
        <div
          className="row"
        >
          <div
            className="col s12 center"
          >
            <h3>
              <i
                className="mdi-content-send brown-text"
              />
            </h3>
            <h4>
              About Us
            </h4>
            <p
              className="left-align light"
            >
              The railway reservation system facilitates the passengers to enquiry about the trains available on the basis of source and destination, booking and cancellation of tickets, enquiry about the status of the booked ticket, etc
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="parallax-container valign-wrapper"
    >
      <div
        className="section no-pad-bot"
      >
        <div
          className="container"
        >
          <div
            className="row center"
          >
            <h5
              className="header col s12 light"
            >
              Railway Booking make Easy
            </h5>
          </div>
        </div>
      </div>
      <div
        className="parallax"
      >
        <img
          alt="Unsplashed background img 3"
          src="images/background3.jpg"
        />
      </div>
    </div>
    <footer
      className="page-footer teal"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col l6 s12"
          >
            <h5
              className="white-text"
            >
              Railway
            </h5>
            <p
              className="grey-text text-lighten-4"
            >
              Rail transport is also known as train transport. It is a means of transport, on vehicles which run on tracks (rails or railroads). It is one of the most important, commonly used and very cost effective modes of commuting and goods carriage over long, as well as, short distances.
            </p>
          </div>
          <div
            className="col l3 s12"
          >
            <h5
              className="white-text"
            >
              Settings
            </h5>
            <ul>
              <li>
                <a
                  className="white-text"
                  href="#!"
                >
                  Reserve Tickets
                </a>
              </li>
              <li>
                <a
                  className="white-text"
                  href="#!"
                >
                  Download Tickets
                </a>
              </li>
            </ul>
          </div>
          <div
            className="col l3 s12"
          >
            <h5
              className="white-text"
            >
              Connect
            </h5>
            <ul>
              <li>
                <a
                  className="white-text"
                  href="#!"
                >
                  Admin
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="footer-copyright"
      >
        <div
          className="container"
        >
          Made for
           
          <a
            className="brown-text text-lighten-3"
            href="/"
          >
            Railway Reservation
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
